### PR TITLE
SPEED UP!

### PR DESCRIPTION
**Pull request by R46bit**

### CODE REVIEW

File: gover/proxy/bypass.go [hunk 1]
```diff
@@ -15,17 +15,17 @@ func genPrngSeed(seed uint64) uint64 {
...
-		}
-		return false
+		k.Seed(seed)
+		k.Seed(k.Int64())
+		k.Int64()
+		// 0     2     4     6     |       10
+		// 45 67 ?? ?? 00 ?? 00 00 | ?? ?? ?? ??...
+		// ^~~~~ ^~~~~ ^~~~~ ^~~~~~~~~~~~~ ^~~~~~...
+		// MAGIC CMDID H_LEN B_LEN         H_DATA...
+		return v^k.Int64()&0xFFFF0000FF00FFFF == 0x4567000000000000
```
Comment (Night12138) [inline gover/proxy/bypass.go]: It's a very sensible approach, but I had tried to match magic only the beginning of the payload before, and there was a certain chance of failure, so I added a matching magic at the end; is there a way to prove that only detecting the beginning is feasible?

Comment (Night12138) [reply gover/proxy/bypass.go]: FYI, the header length and body length may can be calculated, according to past experience, the header length is always fixed

**Comment by Night12138** [comment]
btw, the most time-consuming part of this method is actually generating the first `Int64()`, so if still need to initialize an MT64, maybe it will not reduce too much overhead

**Comment by R46bit** [comment]
> is there a way to prove that only detecting the beginning is feasible?

it's enough for PlayerLoginReq and PlayerLoginRsp packets in synchronous flow

> btw, the most time-consuming part of this method is actually generating the first `Int64()`, so if still need to initialize an MT64, maybe it will not reduce too much overhead

also removed encryption key gen to reduce mem alloc, copypaste packet, and xor the entire packet

**Comment by Night12138** [comment]
LGTM

**Pull request merged**